pkg/operator/ceph/cluster/controller.go

```diff
@@ -186,6 +188,10 @@ func (c *ClusterController) StopWatch() {
 	c.clusterMap = make(map[string]*cluster)
 }
 
+func (c *ClusterController) GetClusterCount() int {
+	return len(c.clusterMap)
```

Comment:
If we keep the lock would we need to add the lock here? Or is the caller of `GetClusterCount()` already in the lock?

Reply:
The caller of GetClusterCount is in the lock (it's locked before the deletion callbacks are called and released after). It's true that this is an exported function, and I don't love that being left unlocked if we leave the locks in at all, but (assuming the locks have any utility, TBD below), we need to put the lock around the map access and the subsequent CSI resource creation or deletion.